### PR TITLE
[Windows] Fix Span Foreground property

### LINKED ISSUE #23488: Span text-decoration is incorrect whereas the Label behaves properly
### Description

Span text-decoration is incorrect whereas the Label behaves properly.

Have created a Label with and without FormattedText.

And applied similar text decorations on both of them.

The Label styled the text and its underlining using the color defined whereas the underlying Span of the FormattedText didn't adapt the color defined for underlining.

### Steps to Reproduce

1. Create a new .NET MAUI App
2. Replace the MainPage content with the below snippet
3. Build and run the app to see the difference in text formatting

```xml
<StackLayout
    HorizontalOptions="Center"
    VerticalOptions="Center">
    <Label
        Text="hyperlink"
        TextColor="Blue"
        TextDecorations="Underline" />
    <Label>
        <Label.FormattedText>
            <FormattedString>
                <Span
                    Text="hyperlink"
                    TextColor="Blue"
                    TextDecorations="Underline" />
            </FormattedString>
        </Label.FormattedText>
    </Label>
</StackLayout>
```

### Link to public reproduction project repository

Plain control definition, default template is sufficient

### Version with bug

8.0.61 SR6.1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

Windows SDK 10.0.19041

### Did you find any workaround?

None.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Vignesh-SF3580**
### Root Cause
The label uses the run element in WinUI platform while using formatted text. The label text color is applied to the text highlighter of the run element, which updates the run text color, but to update the underline color of hypertext, the run Foreground property needs to be updated, which was missed.
 
### Description of Issue Fix
The fix involved updating the run Foreground property to ensure the underline color of hypertext is also updated, preventing it from remaining unchanged from the default.

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/23488

### Screenshots

Before
image

After
image

### CODE REVIEW

**Comment by PureWeen** [comment]
Works great!!

Once the tests are added this should be good to merge

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue23488.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+		{
+		}
+
+		public override string Issue => "Span text-decoration is incorrect whereas the Label behaves properly";
+
+		[Test]
+		[Category(UITestCategories.Label)]
+		public async Task LabelHyperlinkUnderlineColor()
+		{
+			await Task.Delay(500);
```
Comment (PureWeen) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue23488.cs:19]: Add an automation ID to the `Label` here https://github.com/dotnet/maui/pull/24063/files#diff-b1f8b9006e0c54ae7bc49f411b3594a69604ede70d85508200debae016192994R8 and then use `App.WaitForElement` on that AutomationId

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue23488.cs (hunk 1 above)
Comment (Vignesh-SF3580) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue23488.cs]: Added automation ID to the label.

**Review by PureWeen — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]

**Review by Foda — APPROVED** [review]

**Comment by Vignesh-SF3580** [comment]
@dotnet-policy-service agree

**Pull request merged**